### ISSUE TITLE
XLA SPMD causes each process to spawn independently

### ISSUE THREAD

**Issue by alexanderswerdlow** [huggingface/accelerate#2988]
### System Info

```Shell
- `Accelerate` version: 0.33.0
- Platform: Linux-5.19.0-1022-gcp-x86_64-with-glibc2.35
- Python version: 3.10.12
- Numpy version: 1.26.4
- PyTorch version (GPU?): 2.4.0+cu121 (False)
- PyTorch XPU available: False
- PyTorch NPU available: False
- PyTorch MLU available: False
- PyTorch MUSA available: False
- System RAM: 400.47 GB
```

### Information

- [ ] The official example scripts
- [X] My own modified scripts

### Tasks

- [ ] One of the scripts in the examples/ folder of Accelerate or an officially supported `no_trainer` script in the `examples` folder of the `transformers` repo (such as `run_no_trainer_glue.py`)
- [X] My own task or dataset (give details below)

### Reproduction

Run with XLA_USE_SPMD=1, e.g:

```
XLA_USE_SPMD=1 PJRT_DEVICE=TPU accelerate launch --tpu --num_processes=4 --main_training_function=main main.py ...
```

This spawns 4 processes, each of which report a world size of 1. I believe this is related to this [function](https://github.com/pytorch/xla/blob/7135a7e95b05d3d92a7e80c46394ef2595bb93bc/torch_xla/core/xla_model.py#L143), where XLA reports all devices to the same. This means that in my training loop, the step counts are out of sync because it appears there are 4 separate training runs instead of one with synced gradients.

### Expected behavior

The use of SPMD is handled and a single training run occurs.

**Comment by alexanderswerdlow** [comment]
It's not very clear from the documentation (xla or accelerate), but when using SPMD, you should only spawn one process per host, so closing this issue!
